Add new requirements for stub in module guidelines (#2621)
nf-core/website: sites/docs/src/content/docs/guidelines/components/modules.md

Alternate suggestions include using `grep -c` to search for a valid string match, or other tool which will appropriately error when the expected output is not successfully created.

### Stubs

#### Stub block must exist

A stub block MUST exist for all modules

#### Stub files for all output channels 

> The stub block MUST include the creation of a file for every output channel (both mandatory and optional), generated with touch, e.g.

Author: ```suggestion
The stub block MUST include the creation of at least one file for every output channel (both mandatory and optional), generated with touch, e.g.
```

```bash
touch ${prefix}.txt

> ```

Reviewer: I would add that the sub block should reproduce as much as possible the number and filenames that are expected as output

Author: Done

> 

Author: ```suggestion

Ideally, the stub block should reproduce as much as possible the number of, and filenames, of the files expected as output.

```

Alternate suggestions include using `grep -c` to search for a valid string match, or other tool which will appropriately error when the expected output is not successfully created.

### Stubs

#### Stub block must exist

> A stub block MUST exist for all modules

Reviewer 2: ```suggestion
[A stub block](https://www.nextflow.io/docs/latest/process.html#stub) MUST exist for all modules. This is a block of code that replaces the `script` command when the option `-stub` is set. This enables quick testing of the workflow logic, as a "dry-run".
```

#### Stub files for all output channels 

The stub block MUST include the creation of at least one file for every output channel (both mandatory and optional), generated with touch, e.g.

> ```bash
> touch ${prefix}.txt
> ```

Reviewer 2: ```suggestion
```bash
stub:
"""
touch ${prefix}.txt
"""
```
```

Ideally, the stub block should reproduce as much as possible the number of, and filenames structure, of the files expected as output.

#### Stub gzip files must use echo and pipe

Stub files that should be output as gzip compressed, MUST use the syntax in the following example:

```bash
echo "" | gzip > ${prefix}.txt.gz
```

> :::info{title="Rational" collapse}

Reviewer 3: ```suggestion
:::info{title="Rationale" collapse}
```

Alternate suggestions include using `grep -c` to search for a valid string match, or other tool which will appropriately error when the expected output is not successfully created.

### Stubs

#### Stub block must exist

[A stub block](https://www.nextflow.io/docs/latest/process.html#stub) MUST exist for all modules. This is a block of code that replaces the `script` command when the option `-stub` is set. This enables quick testing of the workflow logic, as a "dry-run".

> 
> #### Stub files for all output channels 

Author: ```suggestion

#### Stub block prefix and versions

The stub block MUST include the same variables (e.g. `prefix`) and HEREDOC code as the main script block. 

#### Stub files for all output channels 
```

#### Stub block must exist

[A stub block](https://www.nextflow.io/docs/latest/process.html#stub) MUST exist for all modules. This is a block of code that replaces the `script` command when the option `-stub` is set. This enables quick testing of the workflow logic, as a "dry-run".

#### Stub files for all output channels 

The stub block MUST include the creation of at least one file for every output channel (both mandatory and optional), generated with touch, e.g.

```groovy
stub:
"""
touch ${prefix}.txt

> """

Reviewer 3: ````suggestion
"""
```
````
